src/Maestro/Maestro.Web/MaestroBarClient.cs

```diff
@@ -0,0 +1,328 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Maestro.Data;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Kusto;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.TeamFoundation.Build.WebApi.Events;
+using System;
+using System.Data;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Maestro.Web
+{
+    /// <summary>
+    ///     A bar client interface for use by DarcLib which talks directly
+    ///     to the database for diamond dependency resolution.  Only a few features are required.
+    /// </summary>
+    internal class MaestroBarClient : IBarClient
```

Comment:
In general, we haven't abbreviated "BAR" in the source code anywhere... we've always called it BuildAssetRegistry.  We should probably maintain that pattern (I don't think many pieces of code have the type name in it, so it's not that bad if it's a bit long)

Reply:
Everywhere we have a MaestroBarClient, it's called MaestroBarClient. So like, SubscriptionActorService has a MaestroBarClient (this is what I used as a starting point).

Reply:
Didn't realize there was another. This is good. :-)